src/NServiceBus.Core/Transports/Msmq/QueuePermissions.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace NServiceBus.Features
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Messaging;
+    using System.Security;
+    using Logging;
+
+    static class QueuePermissions
+    {
+        public static void CheckQueuePermissions(IReadOnlyCollection<string> queues)
+        {
+            foreach (var address in queues)
+            {
+                CheckQueue(address);
+            }
+        }
+
+        static void CheckQueue(string address)
+        {
+            var msmqAddress = MsmqAddress.Parse(address);
+
+            MessageQueue messageQueue;
+            if (!MsmqUtilities.TryOpenQueue(msmqAddress, out messageQueue))
+            {
+                Logger.Warn($"Unable to open the queue at address '{address}'. Make sure the queue exists, and the address is correct.");
+                Logger.Debug("Processing will still continue.");
+                return;
```

Comment:
Given how this is used in `MsmqTransportInfrastructure` does this behavior make sense? Do we want to just log a warning and keep going? If the transport can't open it's own queue, shouldn't we throw here?